src/glibc.zig

```diff
@@ -809,6 +849,23 @@ pub fn buildSharedObjects(comp: *Compilation, prog_node: std.Progress.Node) !voi
         var opt_symbol_name: ?[]const u8 = null;
         var versions_buffer: [32]u8 = undefined;
         var versions_len: usize = undefined;
+
+        // There can be situations where there are multiple inclusions for the same symbol with
+        // overlapping versions. For example:
```

Comment:
seems like a problem with the data, not the unpacking logic, right?

Reply:
Er, I definitely should have made this clear in the comment: The two `lgammal` inclusions in question have different target lists.

Here's the relevant `list_symbols.zig` output:

```
 lgammal:
  library: libm.so
  versions: 2.4 2.23
  targets: arm-linux-gnueabi armeb-linux-gnueabi arm-linux-gnueabihf armeb-linux-gnueabihf sparc-linux-gnu powerpc-linux-gnueabi powerpc-linux-gnueabihf powerpc64-linux-gnu s390x-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.32
  targets: arc-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.29
  targets: csky-linux-gnueabi csky-linux-gnueabihf
 lgammal:
  library: libm.so
  versions: 2.0 2.23
  targets: m68k-linux-gnu mipsel-linux-gnueabihf mips-linux-gnueabihf mipsel-linux-gnueabi mips-linux-gnueabi x86-linux-gnu sparc-linux-gnu powerpc-linux-gnueabi powerpc-linux-gnueabihf mips64el-linux-gnuabi64 mips64-linux-gnuabi64 mips64el-linux-gnuabin32 mips64-linux-gnuabin32
 lgammal:
  library: libm.so
  versions: 2.33
  targets: riscv32-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.17 2.23
  targets: powerpc64le-linux-gnu aarch64-linux-gnu aarch64_be-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.3
  targets: powerpc64-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.2.5 2.23
  targets: x86_64-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.16 2.23
  targets: x86_64-linux-gnux32
 lgammal:
  library: libm.so
  versions: 2.27
  targets: riscv64-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.2 2.23
  targets: sparc64-linux-gnu s390x-linux-gnu
 lgammal:
  library: libm.so
  versions: 2.36
  targets: loongarch64-linux-gnu
```

Reply:
So, based on my understanding of [the intent here](https://github.com/ziglang/glibc-abi-tool?tab=readme-ov-file#inclusions), I *think* the data is fine, probably?

If you agree, I'll update the comment to be a bit clearer on this.

Reply:
Thanks, this makes sense now.